src/parser.rs

```diff
@@ -445,6 +448,58 @@ impl Parser<'_> {
                 location: keyword_location,
             }
             .into())
+        } else if keyword == self.if_keyword {
+            let mut arms = Vec::new();
+
+            // if
+            let location = keyword_location;
+            let condition = self.parse_expression(current_query)?;
+            self.consume_whitespace();
+            let statements = self.parse_statements(current_query)?;
+            self.consume_whitespace();
+            arms.push(ast::ConditionalArm {
+                condition,
+                statements,
+                location,
+            });
```

Comment:
Would it perhaps make sense to factor this out into its own function? The repetition in all three cases (`if`/`elif`/`else`) is a bit much. (Though of course the `else` branch is special.)

Reply:
I've tried a few times to reduce the duplication, but so far without success. The small differences always get in the way of abstracting a substantial part of those blocks.

Reply:
Yeah that's what I figured.

At any rate, I'm planning to replace the parser with a custom `tree-sitter` grammar at some point anyway, so it doesn't really matter how slick the current parser is.